internal/schedules/filter.go

```diff
@@ -37,6 +39,23 @@ func NoRouteHandler(krakens map[string]kraken.Kraken, publisher Publisher) gin.H
 				c.JSON(http.StatusNotFound, gin.H{"message": "at least one filter is required"})
 				return
 			}
+			direction_type := c.QueryArray("direction_type")
+			if len(direction_type) > 0 {
+				dt := direction_type[0]
+				directions := []string{"backward", "forward", "all"}
+				if !utils.Contains(directions, dt) {
+					c.JSON(http.StatusBadRequest,
+						gin.H{"message": fmt.Sprintf("wrong direction type parameter selected : %s, it should be [forward, backward, all]]", dt)})
+					return
+				}
+				if dt != "all" {
+					if dt == "forward" {
+						request.Filters = append(request.Filters, "route.has_direction_type(forward,clockwise,inbound)")
+					} else {
+						request.Filters = append(request.Filters, "route.has_direction_type(backward,anticlockwise,outbound)")
+					}
+				}
```

Comment:
I believe that the following is more straightforward and equivalent? Because the code you propose, it seems there is a branch exclusively for `backward` but you can't see it unless you understand that it's not `all`, then it's not `forward` and you are in the `else`

```suggestion
				if dt == "forward" {
					request.Filters = append(request.Filters, "route.has_direction_type(forward,clockwise,inbound)")
				}
				if dt == "backward" {
					request.Filters = append(request.Filters, "route.has_direction_type(backward,anticlockwise,outbound)")
				}
```

Reply:
Done